streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java

```diff
@@ -185,8 +195,10 @@ public void consumerConfigShouldContainAdminClientConfigsForRetriesAndRetryBackO
     public void testGetMainConsumerConfigsWithMainConsumerOverridenPrefix() {
         props.put(StreamsConfig.consumerPrefix(ConsumerConfig.MAX_POLL_RECORDS_CONFIG), "5");
         props.put(StreamsConfig.mainConsumerPrefix(ConsumerConfig.MAX_POLL_RECORDS_CONFIG), "50");
+        props.put(StreamsConfig.mainConsumerPrefix(ConsumerConfig.GROUP_ID_CONFIG), "another-id");
```

Comment:
Should this be `GROUP_INSTANCE_ID_CONFIG` ?

Reply:
This is a piggy-backed test coverage not directly related to the PR --- I wanted to make sure that whatever user specified group id is always overridden by the streams itself.

Reply:
Ack.

Should we test that the `GROUP_INSTANCE_ID` is set correctly based on the user config for the main consumer (or do we have a test for this already?)